microsoft-edge/devtools-guide-chromium/evaluate-performance/index.md

```diff
@@ -224,32 +235,31 @@ Finally, we read `m.offsetTop` again, to adjust the direction of the icon.
+// Read the direction.
+const direction = element.dataset.dir;
+// Calculate the next position.
+let nextPos = direction === 'up' ? currentPos - (SPEED * 100 / canvasHeight) : currentPos + (SPEED * 100 / canvasHeight);
+
+// If the new position is out of bounds, reset it, and switch the direction.
+if (nextPos < 0) {
+  nextPos = 0;
+  element.dataset.dir = 'down';
+} else if (nextPos > 100) {
+  nextPos = 100;
+  element.dataset.dir = 'up';
 }
-```
-
-In the optimized version, we first set the value of the `pos` variable by reading `m.style.top` instead of using `m.offsetTop`.  Using the element's inline style is faster, because reading `m.offsetTop` forces the browser engine to know where all the elements are on the page, which requires the engine to recalculate the styles and the layout.
-
-We then set the new position of the icon, like done in the unoptimized version, but we don't read `m.offsetTop` again (like done in the unoptimized version) to adjust the icon's direction.
 
-The unoptimized code reads and writes the position of the icon from two different places, forcing the browser to recalculate the style and layout on each frame. The optimized version, however, writes and reads the position of the icon in the inline styles only.
+// Set the new position on the element.
+element.style.transform = `translateY(${nextPos}vh)`;
+```
 
+In the optimized version, we first set the value of the `nextPos` variable by reading `element.style.transform` instead of using `element.offsetTop`.  Using the element's inline style is faster, because reading `element.offsetTop` forces the browser engine to know where all the elements are on the page, which requires the engine to recalculate the styles and the layout.
 
-<!-- ------------------------------ -->
-#### Further possible optimizations
+We then adjust the direction of the icon, but this time we don't read `element.offsetTop` again like done in the unoptimized version.
 
-This code could be made even faster by using CSS properties that only require the browser to do compositing, rather than layout. Instead of manipulating the `top` property, which forces the browser to run its layout code again, using the `transform` property would allow the browser to consider each icon as an individual layer, and then display these layers in the correct positions by re-compositing the final image. For example, instead of using `m.style.top = pos + "px";`, we can use `m.style.transform = translateY(pos + "px,");`.
+Finally, we set the new position of the icon, but this time we use `element.style.transform` instead of `element.style.top`. Using `element.style.transform` is faster because the CSS `transform` property can be applied by the browser rendering engine without having to recalculate the layout of the page. When using the `transform` property, the browser considers each icon as an individual layer, and then displays these layers in the correct positions by re-compositing the final image.
```

Comment:
```suggestion
Finally, we set the new position of the icon, but this time we use `element.style.transform` instead of `element.style.top`. Using `element.style.transform` is faster, because the CSS `transform` property can be applied by the browser rendering engine without having to recalculate the layout of the page. When using the `transform` property, the browser considers each icon as an individual layer, and then displays these layers in the correct positions by re-compositing the final image.
```